tests/chainer_tests/functions_tests/array_tests/test_as_strided.py

```diff
@@ -0,0 +1,354 @@
+import unittest
+
+from chainer import testing, Variable, grad
+
+import numpy as np
+
+from chainer.functions import as_strided
+from chainer.functions.array.as_strided import _stride_array
+
+
+@testing.parameterize(
+    {'dtype': np.float16},
+    {'dtype': np.float32},
+    {'dtype': np.float64},
+    {'dtype': np.int16},
+    {'dtype': np.int32},
+    {'dtype': np.int64}
+)
+class TestStrideArray(unittest.TestCase):
+    def test_flip_cpu(self):
```

Comment:
How about following code to reduce duplicated?

```python
from chainer.backends import cuda

...

def check_flip(self, xp):
    x = xp.arange(4, dtype=self.dtype)
    y = _stride_array(x, (4,), (-1,), 3)  # [3, 2, 1, 0]
    y_expected = np.flip(x, axis=0)
    testing.assert_allclose(y, y_expect)

def test_flip_cpu(self):
    self.check_flip(np)

@testing.attr.gpu
def test_flip_gpu(self):
    check_flip(self, cuda.cupy)
```

Reply:
This is related to the comment above. I agree with the comment that I should avoid duplication.

Reply:
@okuta please disregard my comment above. fixed.